generator/generator_tests/altitude_test.cpp

```diff
@@ -0,0 +1,216 @@
+{
+public:
+  using TMockAltitudes = map<m2::PointI, TAltitude>;
+
+  MockAltitudeGetter(TMockAltitudes && altitudes) : m_altitudes(altitudes) {}
+
+  // IAltitudeGetter overrides:
+  TAltitude GetAltitude(m2::PointD const & p) override
+  {
+    m2::PointI const rounded(static_cast<int32_t>(round(p.x)), static_cast<int32_t>(round(p.y)));
+    auto const it = m_altitudes.find(rounded);
+    if (it == m_altitudes.end())
+      return kInvalidAltitude;
+
+    return it->second;
+  }
+
+private:
+  TMockAltitudes const & m_altitudes;
+};
+
+vector<m2::PointD> ExtractPoints(TPoint3DList const & geom3D)
+{
+  vector<m2::PointD> result;
+  for (Point3D const & p : geom3D)
+    result.push_back(m2::PointD(p.m_point));
+  return result;
+}
+
+void FillAltitudes(vector<TPoint3DList> const & roads,
+                   MockAltitudeGetter::TMockAltitudes & altitudes)
+{
+  for (TPoint3DList const & geom3D : roads)
+  {
+    for (size_t i = 0; i < geom3D.size(); ++i)
+    {
+      auto it = altitudes.find(geom3D[i].m_point);
+      if (it != altitudes.end())
+      {
+        CHECK_EQUAL(it->second, geom3D[i].m_altitude, ("Point", it->first, "is set with two different altitudes."));
+        continue;
+      }
+      altitudes[geom3D[i].m_point] = geom3D[i].m_altitude;
+    }
+  }
+}
+
+void BuildMwmWithoutAltitudes(vector<TPoint3DList> const & roads, LocalCountryFile & country)
+{
+  generator::tests_support::TestMwmBuilder builder(country, feature::DataHeader::country);
+
+  for (TPoint3DList const & geom3D : roads)
+    builder.Add(generator::tests_support::TestStreet(ExtractPoints(geom3D), string(), string()));
+}
+
+void TestAltitudes(MwmValue const & mwmValue, string const & mwmPath, MockAltitudeGetter & altitudeGetter)
+{
+  AltitudeLoader loader(mwmValue);
+
+  auto processor = [&altitudeGetter, &loader](FeatureType const & f, uint32_t const & id)
+  {
+    f.ParseGeometry(FeatureType::BEST_GEOMETRY);
+    size_t const pointsCount = f.GetPointsCount();
+    TAltitudes const altitudes = loader.GetAltitudes(id, pointsCount);
+
+    if (!routing::IsRoad(feature::TypesHolder(f)))
+    {
+      TEST(altitudes.empty(), ());
+      return;
+    }
+
+    TEST_EQUAL(altitudes.size(), pointsCount, ());
+
+    for (size_t i = 0; i < pointsCount; ++i)
+      TEST_EQUAL(altitudeGetter.GetAltitude(f.GetPoint(i)), altitudes[i], ("A wrong altitude"));
+  };
+  feature::ForEachFromDat(mwmPath, processor);
+}
+
+void TestAltitudesBuilding(vector<TPoint3DList> const & roads)
+{
+  classificator::Load();
+  Platform & platform = GetPlatform();
+  string const testDirFullPath = my::JoinFoldersToPath(platform.WritableDir(), kTestDir);
+
+  // Building mwm without altitude section.
+  LocalCountryFile country(testDirFullPath, CountryFile(kTestMwm), 1);
+  platform::tests_support::ScopedDir testScopedDir(kTestDir);
+  platform::tests_support::ScopedFile testScopedMwm(country.GetPath(MapOptions::Map));
+  UNUSED_VALUE(testScopedDir);
```

Comment:
To prevent compilation warning on all our platforms. I'm removing them and we'll check if everything is ok.